src/tests/end-to-end/common/browser-factory.ts

```diff
@@ -17,9 +17,11 @@ const fileExists = util.promisify(fs.exists);
 const writeFile = util.promisify(fs.writeFile);
 const readFile = util.promisify(fs.readFile);
 
+export type ExtraPermissions = 'localhost' | 'all-origins';
```

Comment:
It may not be clear to an unfamiliar reader why they'd ever want to use `localhost`, given that the only `permissions.request` calls we'll have in the product will be for all-origins; maybe naming something like `fake-activeTab` or something would be more clear, possibly with a comment explaining and/or linking the puppeteer issue?

Reply:
updated